### PR TITLE
8.0 sms update

### PR DESCRIPTION
**Pull request by ahilali**
Make all fields readonly when message was sent
@sebastienbeau

### CODE REVIEW

File: sms_send_picking/README.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+SMS send picking
+===========
+
+This module provide a cron and a method to automaticly sending a SMS when a
```
Comment (elicoidal) [inline sms_send_picking/README.rst]: s/provide/provides

File: sms_send_picking/README.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+picking delivery is ready to transfert
```
Comment (elicoidal) [inline sms_send_picking/README.rst]: period

File: sms_send_picking/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline sms_send_picking/__openerp__.py:1]: simplify headers following https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: sms_send_picking/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+#   GNU Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+
+{
+    'name': 'Sms Send Picking',
+    'version': '0.0.1',
```
Comment (elicoidal) [inline sms_send_picking/__openerp__.py]: Following versioning guidelines: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md

File: sms_send_picking/stock.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    availability_sent_by_sms = fields.Boolean(default=False)
+
+    #TODO use a templating instead
+    @api.model
+    def _prepare_availability_by_sms_notification(self):
+        gateway = self.env['sms.gateway'].search([
+            ('default_gateway', '=', True)], limit=1)
+        return {
+            'gateway_id': gateway.id,
+            'message': _('Your picking %s is ready to transfert') % self.name,
```
Comment (elicoidal) [inline sms_send_picking/stock.py]: s/transfert/transfer
add period

File: smsclient_core/README.rst [hunk 6]
```diff
@@ -0,0 +1,72 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
```
Comment (elicoidal) [inline smsclient_core/README.rst:1]: We should have one commit per module.

File: smsclient_core/README.rst [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    :alt: License: AGPL-3
+
+SMS client core
+===========
+
+This module provide base function to send sms. It not works in stand alone, it needs a least one provider module
```
Comment (elicoidal) [inline smsclient_core/README.rst]: s/provide/provides

Comment (elicoidal) [inline smsclient_core/README.rst]: s/It not works in stand alone, it needs/It is not meant to work alone and needs
add period

File: smsclient_core/__openerp__.py [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    "name": "SMS Client Core",
+    "version": "1.0",
```
Comment (elicoidal) [inline smsclient_core/__openerp__.py]: Version

File: smsclient_core/serveraction.py [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+_logger = logging.getLogger('gateway')
+
+
+class ServerAction(models.Model):
+    """
+    Possibility to specify the SMS Gateway when configure this server action
+    """
+    _inherit = 'ir.actions.server'
+
+    mobile = fields.Char('Mobile No', help="Provides fields that "
```
Comment (elicoidal) [inline smsclient_core/serveraction.py]: Field to be used to fetch the mobile number. eg: you select the Invoice model and `object.invoice_address_id.mobile` will be the field providing the correct mobile number.

File: smsclient_core/serveraction.py [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+                         "be used to fetch the mobile number, e.g. you select"
+                         " the invoice, then "
+                         "`object.invoice_address_id.mobile` is the field "
+                         "which gives the correct mobile number")
+    sms = fields.Char('SMS', size=160, translate=True)
+    sms_server = fields.Many2one('sms.gateway', 'SMS Server',
+                                 help='Select the SMS Gateway configuration'
+                                 ' to use with this action')
```
Comment (elicoidal) [inline smsclient_core/serveraction.py]: period

File: smsclient_core/serveraction.py [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+    sms_template_id = fields.Many2one('email.template', 'SMS Template',
+                                      help='Select the SMS Template'
+                                      ' configuration to use with this action')
```
Comment (elicoidal) [inline smsclient_core/serveraction.py]: period

File: smsclient_core/serveraction.py [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+
+#    @api.model
```
Comment (elicoidal) [inline smsclient_core/serveraction.py]: delete dead code

File: smsclient_core/sms_gateway.py [hunk 13]
```diff
@@ -0,0 +1,269 @@
...
+    login_provider_visible = fields.Boolean(default=False)
+    password_provider = fields.Char(compute='_get_provider_conf')
+    password_provider_visible = fields.Boolean(default=False)
+    from_provider = fields.Char(compute='_get_provider_conf')
+    from_provider_visible = fields.Boolean(default=False)
+    code = fields.Char('Verification Code')
+    code_visible = fields.Boolean(default=False)
+    body = fields.Text('Message',
+                       help="The message text that will be send along with the"
+                            " email which is send through this server")
```
Comment (elicoidal) [inline smsclient_core/sms_gateway.py]: period

File: smsclient_core/sms_gateway.py [hunk 14]
```diff
@@ -0,0 +1,269 @@
...
+    validity = fields.Integer(
+        help='The maximum time - in minute(s) - before the message is dropped',
```
Comment (elicoidal) [inline smsclient_core/sms_gateway.py]: period

File: smsclient_core/sms_gateway.py [hunk 15]
```diff
@@ -0,0 +1,269 @@
...
+             'lenght) or 2 for unicode (70 characters max'
+             'lenght)',
+        default='1'
+    )
+    coding_visible = fields.Boolean(default=False)
+    tag = fields.Char('Tag', help='an optional tag')
+    tag_visible = fields.Boolean(default=False)
+    nostop = fields.Boolean(
+        help='Do not display STOP clause in the message, this requires that '
+             'this is not an advertising message',
```
Comment (elicoidal) [inline smsclient_core/sms_gateway.py]: period

File: smsclient_core/sms_gateway.py [hunk 16]
```diff
@@ -0,0 +1,269 @@
...
+        readonly=True,
+        default='draft')
+    error = fields.Text(
+        'Last Error',
+        size=256,
+        readonly=True,
+        states={'draft': [('readonly', False)]})
+    validity = fields.Integer(
+        'Validity',
+        help='The maximum time -in minute(s)- before the message is dropped')
```
Comment (elicoidal) [inline smsclient_core/sms_gateway.py]: period (and following)

**Comment by Viggor** [comment]
Thanks @elicoidal for the review, I have corrected all of this.

**Comment by elicoidal** [comment]
@Viggor Thanks a lot for your contribution to the OCA :)

**Comment by sebastienbeau** [comment]
sorry @ahilali make a mistakeby proposing again this PR. The original one is here
https://github.com/OCA/connector-telephony/pull/55/files

I have merge the work done by @Viggor

**Pull request closed without merging**